src/Compilers/VisualBasic/Portable/Symbols/Source/CustomModifierUtils.vb

```diff
@@ -47,11 +103,21 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
             If Not overriddenParam.CustomModifiers.SequenceEqual(thisParam.CustomModifiers) OrElse
                (overriddenParam.IsByRef AndAlso thisParam.IsByRef AndAlso overriddenParam.CountOfCustomModifiersPrecedingByRef <> thisParam.CountOfCustomModifiersPrecedingByRef) OrElse
-               thisParam.Type <> overriddenParam.Type Then
+               Not thisParam.Type.IsSameType(overriddenParam.Type, TypeCompareKind.IgnoreTupleNames) Then
+
+                Dim thisParamType As TypeSymbol = thisParam.Type
+                Dim overriddenParamType As TypeSymbol = overriddenParam.Type
+
+                If thisParamType.ContainsTuple() AndAlso Not overriddenParam.Type.IsSameType(thisParamType, TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds) Then
+                    Dim names As ImmutableArray(Of String) = VisualBasicCompilation.TupleNamesEncoder.Encode(thisParamType)
+                    overriddenParamType = TupleTypeDecoder.DecodeTupleTypesIfApplicable(overriddenParamType, thisParamType.ContainingAssembly, names)
+                End If
+
                 thisParam = DirectCast(thisParam, SourceParameterSymbolBase).WithTypeAndCustomModifiers(
```

Comment:
On further reflection, I think this is actually ok as-is, because of the name of the method. In C#, the method used is `WithCustomModifiersAndParams`. So it makes sense that it would not replace the type, but that `WithTypeAndCustomModifiers` would.